From bcedaa9b3abd43eb4f596d152b7f20ecd060ce2e Mon Sep 17 00:00:00 2001
From: yuyue <yuyue@mprc.pku.edu.cn>
Date: Thu, 3 Nov 2011 13:49:28 +0800
Subject: [PATCH 071/641] UniCore64: Add union init_thread_union

Fixed error:
arch/unicore64/kernel/built-in.o:(.data+0x8): undefined reference to `init_thread_union'

Signed-off-by: yuyue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/init_task.c |   14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/unicore64/kernel/init_task.c b/arch/unicore64/kernel/init_task.c
index 39628f4..364d7ed 100644
--- a/arch/unicore64/kernel/init_task.c
+++ b/arch/unicore64/kernel/init_task.c
@@ -2,6 +2,20 @@
 #include <linux/sched.h>
 #include <linux/init_task.h>
 #include <linux/mqueue.h>
+#include <linux/hardirq.h>
+
+/**
+ * union init_thread_union - Initial thread structure.
+ *
+ * We need to make sure that this is 8192-byte aligned due to the
+ * way process stacks are handled. This is done by making sure
+ * the linker maps this in the .text segment right after head.S,
+ * and making head.S ensure the proper alignment.
+ *
+ * The things we do for performance..
+ */
+union thread_union init_thread_union __init_task_data = {
+	INIT_THREAD_INFO(init_task) };
 
 /**
  * struct init_signals -
-- 
1.7.9.5

